src/models/imposter.js

```diff
@@ -54,6 +54,9 @@ function create (Protocol, request) {
 
                 result.requests = server.requests;
                 result.stubs = server.stubs();
+                result.stubs.forEach(function (stub) {
```

Comment:
I'm just initializing the state object when creating the protocols and then using it when the stub are being iterated over to make sure they have a reference to the state. I don't know that I have a reference to the protocol when the injected responseHandler/predicate is being called. This is again not so much for saving a JSON so you can repost it later but rather for the case where you want to return different JSONs depending on what your state is. 
Is there an alternate way of achieving the same use cases?
Thanks!